### DISCUSSION TITLE
How to mock iframe resources ?

### DISCUSSION THREAD

**Discussion by souuu** [webdriverio/webdriverio#8105] [Q&A]
Hello,
When I use browser.mock(), it only works on the top level of the application. But for resources where the initiator is coming from an iframe having a different domain, the http call is not intercepted.
Is there a way to handle this case ?
Thank you,

I have tried to use `--disable-features=site-per-process` in `goog:chromeOptions`, but it didn't work.
It was mentioned here: https://github.com/webdriverio/webdriverio/discussions/7528

**Comment by portyu9** [comment]
Hi @souuu 

The `--disable-features=site-per-process` workaround from #7528 was specific to the old Chromium/CDP implementation and I would not use it as the solution today.

The problem in the original setup is that a cross-origin iframe can run in a separate browser process/target, so the old network interception implementation did not necessarily observe requests made by that target.

With current WebdriverIO, first retest this on the latest WDIO/browser combination using the normal `browser.mock()` API and WebDriver BiDi support where available. Do **not** disable Chrome site isolation just to make tests pass.

If the request from the third-party iframe is still not visible to `browser.mock()`, there isn't a supported WDIO configuration switch that guarantees interception of that resource. For a deterministic test, the better approach is to mock it outside the browser, for example through:

- a test HTTP/reverse proxy
- a mock server
- an application-controlled replacement for the iframe URL

That also avoids depending on Chromium's internal iframe/process architecture.

I reviewed the linked #7528 discussion as well; its accepted workaround was historical and the author of this discussion had already confirmed it did not solve their case. There were no screenshots or attachments in #8105.

Please mark this answer as accepted if it helped, thank you.
